app/controllers/ClientController.scala

```
@@ -69,9 +75,40 @@ class ClientController @Inject()(appConfig: AppConfig,
         Future.successful(Ok(views.html.individual.correspondenceDetails(appConfig, correspondenceDetailsForm.correspondenceDetailsForm)))
   }
 
-  val submitIndividualCorrespondenceDetails = TODO
+  val submitIndividualCorrespondenceDetails: Action[AnyContent] = authorisedActions.authorisedAgentAction {
+    implicit user =>
+      implicit request =>
+
+        def createRelationship(account: AgentAccount, reference: SubscriptionReference) = {
+          relationshipService.createClientRelationship(RelationshipModel(account.agentCode.value, reference.cgtRef)).flatMap {
+            case SuccessfulRelationshipResponse => Future.successful(Redirect(routes.ClientController.confirmation(reference.cgtRef)))
+            case _ =>
```

Error should be handled in the connector. Do not return a Future.successful(InternalServerError) let it fail by throwing an error or returning a Future.failed. Logging is also not required here.